frontend-react/src/pages/Login.tsx

```diff
@@ -31,6 +33,10 @@ export const Login = () => {
 
     const onSuccess = (tokens: Tokens | undefined) => {
         oktaAuth.handleLoginRedirect(tokens);
+
+        getUserEmail().then((email) => {
```

Comment:
One thing I thought of was that we could probably just add a subscription to the Okta AuthStateManager:

```tsx
// in TelemetryService.tsx
OKTA_AUTH.authStateManager.subscribe((authState) => {
    const { email } = authState?.idToken?.claims || {};

    if (email) {
        appInsights?.setAuthenticatedUserContext(email);
    }
});
```

Since this handles all auth state changes, it should catch all flows that we need so it could simplify the logic in this pull request:
- Unauthed --> navigate on site--> login
- Authed --> navigate on site
- Authed --> logout --> navigate on site